clang/lib/CodeGen/CGLoopInfo.cpp

```diff
@@ -524,6 +524,48 @@ MDNode *LoopInfo::createMetadata(
     LoopProperties.push_back(MDNode::get(Ctx, Vals));
   }
 
+  if (Attrs.SYCLLoopCoalesceEnable) {
+    LLVMContext &Ctx = Header->getContext();
+    Metadata *Vals[] = {MDString::get(Ctx, "llvm.loop.coalesce.enable")};
+    LoopProperties.push_back(MDNode::get(Ctx, Vals));
+  }
+
+  if (Attrs.SYCLLoopCoalesceNLevels > 0) {
+    LLVMContext &Ctx = Header->getContext();
+    Metadata *Vals[] = {
+        MDString::get(Ctx, "llvm.loop.coalesce.count"),
+        ConstantAsMetadata::get(ConstantInt::get(
+            llvm::Type::getInt32Ty(Ctx), Attrs.SYCLLoopCoalesceNLevels))};
+    LoopProperties.push_back(MDNode::get(Ctx, Vals));
+  }
+
+  if (Attrs.SYCLLoopPipeliningDisable) {
+    LLVMContext &Ctx = Header->getContext();
+    Metadata *Vals[] = {MDString::get(Ctx, "llvm.loop.intel.pipelining.enable"),
```

Comment:
I'm confused right now. 'SYCLLoopPipelining**Disable**' -> 'llvm.loop.intel.pipelining.**enable'**? May be I'll get the logic later, while reading Sema part of the change, but for now after reading doc I didn't get it, so either doc shall be updater or it's a bug.

Reply:
Joe Garvey described why they need this renaming. For Intel FPGA pipelining is enabled by default. So, user may set disable_loop_pipelining attribute only in cases when he wants to turn pipelining off. But this setting may be opposite for other FPGA devices. That's why they decided to name this attribute as PipelineEnableINTEL in SPIR-V spec.

Reply:
However, I think that here should be a comment that this is done on purpose.